client/src/client.ts

```diff
@@ -30,8 +30,17 @@ export class AngularLanguageClient implements vscode.Disposable {
   private readonly outputChannel: vscode.OutputChannel;
   private readonly clientOptions: lsp.LanguageClientOptions;
   private readonly name = 'Angular Language Service';
+  private readonly virtualDocumentContents = new Map<string, string>();
 
   constructor(private readonly context: vscode.ExtensionContext) {
+    vscode.workspace.registerTextDocumentContentProvider('angular-embedded-content', {
+      provideTextDocumentContent: uri => {
+        const originalUri = uri.path.slice(1).slice(0, -1 * '.html'.length);
```

Comment:
I think we should use standard methods from `path` to extract the basename.
If we do that, we should use `uri.fsPath` instead of `uri.path`.
From vscode doc:
```ts
		/**
		 * The string representing the corresponding file system path of this Uri.
		 *
		 * Will handle UNC paths and normalize windows drive letters to lower-case. Also
		 * uses the platform specific path separator.
		 *
		 * * Will *not* validate the path for invalid characters and semantics.
		 * * Will *not* look at the scheme of this Uri.
		 * * The resulting string shall *not* be used for display purposes but
		 * for disk operations, like `readFile` et al.
		 *
		 * The *difference* to the [`path`](#Uri.path)-property is the use of the platform specific
		 * path separator and the handling of UNC paths. The sample below outlines the difference:
		 * ```ts
		const u = URI.parse('file://server/c$/folder/file.txt')
		u.authority === 'server'
		u.path === '/shares/c$/file.txt'
		u.fsPath === '\\server\c$\folder\file.txt'
		```
		 */
		readonly fsPath: string;
```

Reply:
Changed to store the vdoc URI in the map instead of the original URI.